optional_plugins/varianter_pict/avocado_varianter_pict/__init__.py

```diff
@@ -0,0 +1,176 @@
+    """
+
+    name = 'varianter_pict'
+    description = "Pict based Varianter options for the 'run' subcommand"
+
+    def configure(self, parser):
+        try:
+            pict_binary = utils_path.find_command('pict')
+        except utils_path.CmdNotFoundError:
+            pict_binary = None
+
+        for name in ("run", "variants"):  # intentionally ommiting "multiplex"
+            subparser = parser.subcommands.choices.get(name, None)
+            if subparser is None:
+                continue
+            pict = subparser.add_argument_group('pict based varianter options')
+            pict.add_argument('--pict-binary', metavar='PATH',
+                              default=pict_binary,
+                              help=('Where to find the binary version of the '
+                                    'pict tool. Tip: download it from '
+                                    'https://github.com/Microsoft/pict and '
+                                    'run `make` to build it'))
+            pict.add_argument('--pict-parameter-file', metavar='PATH',
+                              help=("Paths to a pict parameter file"))
+            pict.add_argument('--pict-parameter-path', metavar='PATH',
+                              default='/run',
+                              help=('Default path for parameters generated '
+                                    'on the Pict based variants'))
+            pict.add_argument('--pict-order-of-combinations',
+                              metavar='ORDER', type=int, default=2,
+                              help=("Order of combinations. Defaults to "
+                                    "%(default)s, maximum number is specific "
+                                    "to parameter file content"))
+
+    def run(self, args):
+        pass
+
+
+def run_pict(binary, parameter_file, order):
+    cmd = "%s %s /o:%s" % (binary, parameter_file, order)
+    return process.system_output(cmd, shell=True)
+
+
+def parse_pict_output(output):
+    variants = []
+    lines = output.splitlines()
+    header = lines[0]
+    headers = header.split('\t')
+    for line in lines[1:]:
+        variants.append(dict(zip(headers, line.split('\t'))))
+    return (headers, variants)
+
+
+class VarianterPict(Varianter):
+
+    """
+    Processes the pict file into variants
+    """
+
+    name = 'yaml_to_mux'
+    description = 'Multiplexer plugin to parse yaml files to params'
+
+    def initialize(self, args):
+        self.variants = None
+        error = False
+
+        pict_parameter_file = getattr(args, "pict_parameter_file", None)
+        if pict_parameter_file is None:
+            return
+        if not os.access(pict_parameter_file, os.R_OK):
+            LOG_UI.error("pict parameter file '%s' could not be found or is "
+                         "not readable", pict_parameter_file)
+            error = True
+
+        pict_binary = getattr(args, "pict_binary", None)
+        if pict_binary is None:
```

Comment:
This test is specific to `None` value. The message should be nicer.... `None` is internal. Instead of:
```
pict binary 'None' could not be found, please set it with --pict-binary or put it in your PATH
```
I'd suggest something like:
```
Pict binary not found in $PATH. Please set its location with --pict-binary.
```

Reply:
Agreed.